### ISSUE TITLE
Can't access Database after update

### ISSUE THREAD

**Issue by bf8392** [tiredofit/docker-db-backup#49]
Hi :-) I recently made an update to your latest container, and since then I get:
```
Database not accessible - retrieving
```
Are docker-secrets still supported? I'm running in swarm...

**Comment by tiredofit** [comment]
Secrets still should be supported! 
Whats your DB_TYPE ?

**Comment by bd8392** [comment]
Mariadb

**Comment by tiredofit** [comment]
Interesting, We haven't touched that in a while.
Could you send me some full container logs privately to my email?

Preferably DEBUG_MODE=TRUE - Note, you'll want to hide your host and password as that will expose it.

**Comment by bd8392** [comment]
I think there is a typo in the mysql command...there is no space after password...also user seems not to be picked up...If you spot some private Info please inform me, but I think I filtered everything out =)

```
q8soi  [44m[WARN][49m /etc/services.available/10-db-backup/run ** [db-backup] MySQL/MariaDB Server 'mariadb' is not accessible, retrying.. (45 seconds so far)
q8soi  + sleep 5
q8soi  + ((  COUNTER+=5  ))
q8soi  + true
q8soi  + mysqlcmd='mysql -u -P 3306 -h mariadb -pSomepass'
q8soi  ++ mysql -u -P 3306 -h mariadb '-pSomepass' -e 'SELECT COUNT(*) FROM information_schema.FILES;'
q8soi  + out='ERROR 1045 (28000): Access denied for user '\''-P'\''@'\''33.6.3.9'\'' (using password: YES)'
q8soi  + echo 'ERROR 1045 (28000): Access denied for user '\''-P'\''@'\''33.6.3.9'\'' (using password: YES)'
q8soi  + grep -E 'COUNT|Enter'
q8soi  + '[' 1 -eq 0 ']'
q8soi  + print_warn 'MySQL/MariaDB Server '\''mariadb'\'' is not accessible, retrying.. (50 seconds so far)'
q8soi  + output_off
q8soi  + '[' TRUE = TRUE ']'
q8soi  + set +x
```

**Comment by tiredofit** [comment]
No space is correct after the -p.

Are you sure you are on the most recent image? We stripped out that -p field a while back and started using a different way for password storage inside the image. You can try a tagged release `tiredofit/db-backup:2.3.0`

**Comment by bd8392** [comment]
Yes I justz checked...using taged tiredofit 2.3.0...Here is my config:
```
  tiredofit-db-backup:
    image: tiredofit/db-backup:2.3.0
    environment:
      COMPRESSION: NONE
      DB_CLEANUP_TIME: '2880'
      DB_DUMP_FREQ: '1440'
      DB_HOST: mariadb
      DB_PASS_FILE: /database_password
      DB_PORT: '3306'
      DB_TYPE: mariadb
      DB_USER_FILE: /database_user
      DEBUG_MODE: 'TRUE'
      MD5: 'TRUE'
      SPLIT_DB: 'TRUE'
      TIMEZONE: Europe/Vienna
    volumes:
     - mariadb_backups:/backup
     - tiredofit_scripts:/assets/custom-scripts
    networks:
     - apprise_notifier_net
     - database_stack_net
    secrets:
     -
      source: mariadb_root_password
      target: /database_password
     -
      source: mariadb_root_user
      target: /database_user
    logging:
      driver: json-file
    deploy:
      labels:
        swarmpit.service.deployment.autoredeploy: 'true'
```
I can`t figure out qwhats wrong as everything is mounted as before...

**Comment by tiredofit** [comment]
I will try to see if I can recreate on my end.

**Comment by bd8392** [comment]
Okay =). By the way the log-message is still with p field...
I have the database on a different stack, thats why I connect it to the network of it.

**Comment by bd8392** [comment]
2.0.0 WORKS =). At least it executes the custom telegram-script we once created =)

**Comment by tiredofit** [comment]
Wow - ok, so there have been alot of changes since then. Wouldn't be surprised if 2.1.1 works the same. `2.2.0` had a secrets change followed by a fix in `2.2.2`.

**Comment by bd8392** [comment]
Okay wait I'll test...=)

**Comment by bd8392** [comment]
2.1.1 executes the script, but sends weird message -> instead of: 
```
mysql Backup completed on mariadb at 2020-10-19 16:50:08 
```
it says:
```
0 Backup completed on mysql at 2020-10-19

**Comment by tiredofit** [comment]
Ahh right - You may want to check the syntax and update your external script. The script now outputs an exit code of the backup as the first value I believe.

**Comment by bd8392** [comment]
> You may want to check the syntax and update your external script.

Yes =) This is my actual external script:
```
#!/bin/bash
curl -s -X POST -d "{\"urls\":\"tgram:someprivateinfo", \"body\":\"${1} Backup completed on ${2} at ${4} ${5}\", \"title\":\"MariaDB_Backup_Home\"}" -H "Content-Type: application/json" http://apprise:8000/notify
exit 0
```
nevertheless...2.3.0 is not working at all, as it doesn't get access to mariadb...2.0.0 throws exit 0 so I assume it working...

**Comment by tiredofit** [comment]
````
##!/bin/bash

## Example Post Script
## $1=EXIT_CODE (After running backup routine)
## $2=DB_TYPE (Type of Backup)
## $3=DB_HOST (Backup Host)
## #4=DB_NAME (Name of Database backed up
## $5=DATE (Date of Backup)
## $6=TIME (Time of Backup)
## $7=BACKUP_FILENAME (Filename of Backup)
## $8=FILESIZE (Filesize of backup)
## $9=MD5_RESULT (MD5Sum if enabled)

echo "${1} ${2} Backup Completed on ${3} for ${4} on ${5} ${6}. Filename: ${7} Size: ${8} bytes MD5: ${9}" 
```` 

Is the new syntax for the script. I'll go and perform some tests to see why 2.3.0 isn't working for you later on today.

**Comment by bd8392** [comment]
> ```
> ##!/bin/bash
> 
> ## Example Post Script
> ## $1=EXIT_CODE (After running backup routine)
> ## $2=DB_TYPE (Type of Backup)
> ## $3=DB_HOST (Backup Host)
> ## #4=DB_NAME (Name of Database backed up
> ## $5=DATE (Date of Backup)
> ## $6=TIME (Time of Backup)
> ## $7=BACKUP_FILENAME (Filename of Backup)
> ## $8=FILESIZE (Filesize of backup)
> ## $9=MD5_RESULT (MD5Sum if enabled)
> 
> echo "${1} ${2} Backup Completed on ${3} for ${4} on ${5} ${6}. Filename: ${7} Size: ${8} bytes MD5: ${9}" 
> ```
> 
> Is the new syntax for the script. I'll go and perform some tests to see why 2.3.0 isn't working for you later on today.

Okay very cool! Thanks :-)

**Comment by bd8392** [comment]
> Wow - ok, so there have been alot of changes since then. Wouldn't be surprised if 2.1.1 works the same. `2.2.0` had a secrets change followed by a fix in `2.2.2`.

Any news on this? Does the latest container now work too? :-)
